lib/Service/StorageService.php

```diff
@@ -161,6 +166,32 @@ public function getUsersForFileId(int $fileId): array {
 		}, $mountInfos);
 	}
 
+	/**
+	 * @param Node $node
+	 * @return array<File>
+	 */
+	public function getAllFilesInFolder(Node $node): array {
+		if (!$node instanceof Folder) {
+			return [];
+		}
+		$mount = $node->getMountPoint();
+		if ($mount->getNumericStorageId() === null) {
+			return [];
+		}
+		$filesGen = $this->getFilesInMount($mount->getNumericStorageId(), $node->getId(), 0, 0);
+		$files = [];
+
+		foreach ($filesGen as $fileId) {
+			$node = current($this->rootFolder->getById($fileId));
+			if (!$node instanceof File) {
```

Comment:
Should we exclude trash here as well?

Reply:
Can we move files to trash though? I couldn't find it in the move window.

Reply:
okay looks like we can with `occ files:move` but then it is not in the trash so probably unintended.